channeldb/payment_control.go

```diff
@@ -257,6 +279,17 @@ func (p *PaymentControl) updateHtlcKey(paymentHash lntypes.Hash,
 				attemptID)
 		}
 
+		// Make sure the shard is not already failed or settled.
+		v := htlcBucket.Get(htlcFailInfoKey)
+		if v != nil {
```

Comment:
Does it need the intermediate v variable?

Reply:
Nope, fixed!